e2e/app/page/notebook-page.ts

```diff
@@ -0,0 +1,185 @@
+    return this.frame()
+      .then( (iframe) => iframe.waitForSelector(CssSelector.runCellButton, {visible: true}))
+      .then( (butn) => butn.click() );
+  }
+
+  /**
+   * Click Save button in toolbar.
+   */
+  async saveNotebook(): Promise<void> {
+    return this.frame()
+      .then( (iframe) => iframe.waitForSelector(CssSelector.saveNotebookButton, {visible: true}))
+      .then( (butn) => butn.click() );
+  }
+
+  /**
+   * Wait for notebook kernel becomes ready (idle).
+   */
+  async waitForKernelIdle(timeOut: number = 60000): Promise<boolean> {
+    const iconSelector = `${CssSelector.kernelIcon}.kernel_idle_icon`;
+    const notifSelector = '#notification_kernel';
+    const frame = await this.frame();
+    try {
+      await Promise.all([
+        frame.waitForSelector(iconSelector, {visible: true, timeout: timeOut}),
+        frame.waitForSelector(notifSelector, {hidden: true, timeout: timeOut}),
+      ])
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async kernelStatus(): Promise<KernelStatus | null> {
+    const elemt = await this.frame().then( (frame) => frame.waitForSelector(CssSelector.kernelIcon, {visible: true}));
+    const value = await getPropValue(elemt, 'title');
+    await elemt.dispose();
+    Object.keys(KernelStatus).forEach(key => {
+      if (KernelStatus[key] === value) {
+        return key;
+      }
+    })
+    return null;
+  }
+
+  async getKernelName(): Promise<string> {
+    const frame = await this.frame();
+    const elemt = await frame.waitForSelector(CssSelector.kernelName, {visible: true});
+    const value = await getPropValue(elemt, 'textContent');
+    await elemt.dispose();
+    return value;
+  }
+
+  /**
+   * Click Run button in toolbar. It run focused cell and insert a new cell below.
+   * @param {number} cellIndex Code Cell index. (index number starts from 1)
+   * @param {string} code New code.
+   * @return {string} Output string.
+   */
+  async runCodeCell(cellIndex: number, opts: { code?: string, timeOut?: number }): Promise<string> {
+    await this.codeCell.selectCell(cellIndex, opts.code);
+    await this.runButton();
+    await Promise.all([
+      this.waitForKernelIdle(opts.timeOut),
+      this.codeCell.waitForOutput(cellIndex, opts.timeOut),
+    ]);
+    return this.codeCell.waitForOutput(cellIndex);
+  }
+
+
+  // ****************************************************************************
+
+  /**
+   * Active Notebook Command or Edit mode.
+   * @param mode
+   */
+  async toggleMode(mode: Mode): Promise<void> {
+    // Press Esc key to activate command mode
+    if (mode === Mode.Command) {
+      return this.page.keyboard.press('Escape');
+    }
+    // Press Enter key to activate edit mode
+    return this.page.keyboard.press('Enter');
+  }
+
+  /**
+   * Run (Execute) cells using Shift+Enter keys.
+   * Shortcut explanation: Shift+Enter run cells and select below.
+   */
+  async runCellShiftEnter(): Promise<void> {
```

Comment:
Optional: We can combine these three methods by just passing 'Shift'/'Control'. 'Alt' and replacing them in keyboard.down. Or 
Keep these three methods and have each one call a common method that takes the command and do the following:
await this.page.bringToFront();
    await this.page.keyboard.down(command);
    await this.page.keyboard.press('Enter', {delay: 20});
    await this.page.keyboard.up(command);